### PR TITLE
Add polymorphic test\sample

### PR DESCRIPTION
**Pull request by steveharter**
Adding a new test that will be used short-term as a sample until we have doc.

It demonstrates how to support polymorphic serialization (serialization only, not deserialization) using a custom converter.

### CODE REVIEW

File: src/System.Text.Json/tests/Serialization/CustomConverterTests.Polymorphic.cs [hunk 1]
```diff
@@ -147,13 +147,74 @@ public static void PersonConverterPolymorphic()
...
             Person person = JsonSerializer.Deserialize<Person>("null");
             Assert.Null(person);
         }
+
+        // A converter that can serialize an abstract Person type.
+        private class PersonPolymorphicSerializerConverter : JsonConverter<Person>
+        {
+            public override Person Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                throw new NotSupportedException($"Deserializing not supported. Type={typeToConvert}.");
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Serialization/CustomConverterTests.Polymorphic.cs]: >  (serialization only, not deserialization)

Why? Was this meant to showcase why the `Type` parameter is important in `Read(...)`?

**Comment by ahsonkhan** [comment]
Looks like a reasonable test improvement to port to 3.0. cc @ericstj

**Comment by karelz** [comment]
@ahsonkhan this PR went into master (5.0), so let's keep the milestone as 5.0.
We should not reflect wishes in milestones - there is currently no PR against 3.0 (at least it is not linked here).

**Comment by ericstj** [comment]
Right, milestone on PR should reflect where it was merged.  Milestone on an issue should reflect the minimum version it was fixed in.

I support porting this to 3.0.

**Comment by ahsonkhan** [comment]
> this PR went into master (5.0), so let's keep the milestone as 5.0.
We should not reflect wishes in milestones - there is currently no PR against 3.0 (at least it is not linked here).

> Right, milestone on PR should reflect where it was merged. 

I see. Got it. Will keep that in mind going forward.

**Pull request merged**